### PR TITLE
312 add vcs tickets

### PR DESCRIPTION
**Pull request by mbao01**
Merge Order - 4

### CODE REVIEW

**Comment by mbao01** [comment]
@yegor256 please review

File: objects/tickets/tagged_tickets.rb [hunk 1]
```diff
@@ -37,31 +32,31 @@ def notify(issue, message)
...
 Please, make sure @0pdd user is in the \
-[list of collaborators](https://github.com/#{@repo}/settings/collaboration):\
+[list of collaborators](#{@vcs.collaborators_link}):\
 \n\n```#{e.class.name}\n#{e.message}\n#{e.backtrace.join("\n")}\n```"
         )
-      rescue Octokit::Error => e
-        @github.add_comment(
-          @repo, issue,
-          "For some reason I wasn't able to add GitHub labels \
+      rescue StandardError => e
```
Comment (yegor256) [inline objects/tickets/tagged_tickets.rb:56]: @mbao01 it's a BAD practice. we try to catch as little as possible, not as much a possible

**Comment by yegor256** [comment]
@mbao01 almost in each pull request you are trying to weaken our system :) please, don't do this. catching exceptions is a BAD practice in general. read this: https://www.yegor256.com/2015/07/09/catch-if-cant-otherwise.html if you catch, you catch only something VERY specific

**Comment by mbao01** [comment]
> @mbao01 almost in each pull request you are trying to weaken our system :) please, don't do this. catching exceptions is a BAD practice in general. read this: https://www.yegor256.com/2015/07/09/catch-if-cant-otherwise.html if you catch, you catch only something VERY specific

@yegor256 Hahaha... thanks for taking your time to review. Please review again.

File: objects/tickets/tagged_tickets.rb [hunk 2]
```diff
@@ -37,31 +32,31 @@ def notify(issue, message)
...
-        needed.each { |t| @github.add_label(@repo, t, 'F74219') }
-        @github.add_labels_to_an_issue(@repo, issue, tags)
-      rescue Octokit::NotFound => e
-        @github.add_comment(
-          @repo, issue,
-          "I can't create GitHub labels `#{needed.join('`, `')}`. \
-Most likely I don't have necessary permissions to `#{@repo}` repository. \
+        needed.each { |t| @vcs.add_label(t, 'F74219') }
+        @vcs.add_labels_to_an_issue(issue_id, tags)
+      rescue StandardError => e
```
Comment (yegor256) [inline objects/tickets/tagged_tickets.rb:47]: @mbao01 again

**Comment by yegor256** [comment]
@mbao01 there are still places

File: objects/tickets/tagged_tickets.rb (hunk 2 above)
Comment (mbao01) [reply objects/tickets/tagged_tickets.rb]: @yegor256 updated

**Comment by mbao01** [comment]
@yegor256 please review

**Comment by yegor256** [comment]
@rultor merge

**Comment by mbao01** [comment]
@yegor256 weird error! I remember this case was addressed in [pdd:#187](https://github.com/cqfn/pdd/pull/187). Is 0pdd using the latest release of pdd?

**Comment by mbao01** [comment]
@yegor256 try again

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/0pdd#345` is not assigned, can't get performer

**Pull request merged**